drivers/sensor/sensor_shell.c

```diff
@@ -106,6 +113,60 @@ static const char *sensor_attribute_name[SENSOR_ATTR_COMMON_COUNT] = {
 	[SENSOR_ATTR_FF_DUR] = "ff_dur",
 };
 
+/* Forward declaration */
+static void data_ready_trigger_handler(const struct device *sensor,
+				       const struct sensor_trigger *trigger);
+
+/**
+ * @brief This table stores a mapping of string trigger names along with the sensor_trigger struct
+ * that gets passed to the driver to enable that trigger, plus a function pointer to a handler. If
+ * that pointer is NULL, this indicates there is not currently support for that trigger type in the
+ * sensor shell.
+ */
+static const struct sensor_shell {
+	const char *name;
+	sensor_trigger_handler_t handler;
+	struct sensor_trigger trigger;
+} sensor_trigger_table[SENSOR_TRIG_COMMON_COUNT] = {
+	[SENSOR_TRIG_TIMER] = {.name = "timer",
+			       .handler = NULL,
+			       .trigger = {.chan = SENSOR_CHAN_ALL, .type = SENSOR_TRIG_TIMER}},
```

Comment:
These are kinda hard to read, how about creating a helper macro?

```c
#define SENSOR_SHELL_TRIGGER_DATA(_type, _name) \
    [_type] = {                                 \
      .name = _name,                            \
      .handler = NULL,                          \
      .trigger = {                              \
        .chan = SENSOR_CHAN_ALL,                \
        .type = _type,                          \
      },                                        \
    }
```

Then
```c
} sensor_trigger_table[SENSOR_TRIG_COMMON_COUNT] = {
    SENSOR_SHELL_TRIGGER_DATA(SENSOR_TRIG_TIMER, "timer"),
    SENSOR_SHELL_TRIGGER_DATA(SENSOR_TRIG_DATA_READY, "data_ready"),
    ...
}
```

Reply:
Yes, this is much cleaner. Added macro helper.